provider/netperf_test.py

```diff
@@ -37,18 +39,22 @@ def netperf_stress(test, params, vm):
     try:
         n_server.start()
         # Run netperf with message size defined in range.
-        netperf_test_duration = params.get_numeric("netperf_test_duration")
+        test_duration = params.get_numeric("netperf_test_duration")
+        deviation_time = params.get_numeric("deviation_time")
         test_protocols = params.get("test_protocol")
         netperf_output_unit = params.get("netperf_output_unit")
         test_option = params.get("test_option", "")
-        test_option += " -l %s" % netperf_test_duration
+        test_option += " -l %s" % test_duration
         if params.get("netperf_remote_cpu") == "yes":
             test_option += " -C"
         if params.get("netperf_local_cpu") == "yes":
             test_option += " -c"
         if netperf_output_unit in "GMKgmk":
             test_option += " -f %s" % netperf_output_unit
-        t_option = "%s -t %s" % (test_option, test_protocols)
+        for protocol in test_protocols.split():
+            error_context.context("Testing %s protocol" % protocol,
+                                  logging.info)
+            t_option = "%s -t %s" % (test_option, protocol)
```

Comment:
Netperf only runs one type of test no matter how many -t options may be present on the command-line. The last -t global command-line option will determine the test to be run. [Default: TCP_STREAM]

However, `t_option` will be overwrite by `t_option = "%s -t %s"`. If you want to run multiple testname, you should use multiple threads.

Reply:
Hi @PaulYuuu Thanks for your reminder. if so, I will delete these few lines of code, it is not used at present, I thought it could test different netperf types in a loop. Now it seems that I understood it incorrectly, so I deleted it。 Thanks a lot

Reply:
Yep, for most test cases, they will only use a single protocol for netperf testing, if you want to support multiple protocols, you can refer `generic/tests/netperf.py`.